cpu/atmega_common/periph/adc.c

```diff
@@ -52,6 +54,7 @@ int adc_init(adc_t line)
 
     _prep();
 
+#if !defined(CPU_ATMEGA8)
```

Comment:
We should stop to use CPU specific defines. Instead the feature can be checked directly.